src/user_profile.cpp

```diff
@@ -160,9 +169,18 @@ bool UserSignature::signFile(string udf_file)
     }
     char *udf = (char *) calloc(statbuf.st_size, sizeof(char));
     FILE *fp = fopen(udf_file.c_str(), "r");
-    if (UdfToChar(udf, fp, udf_file.c_str(), statbuf))
-        if (signUdf(sk, buf, udf, signed_message, signed_message_len))
-            return createDirectoryTree(pk, sk, signed_message, signed_message_len);
+    if (udfFromFile(udf, fp, udf_file.c_str(), statbuf))
+    {
+        signed_message = new unsigned char[crypto_sign_BYTES + statbuf.st_size];
+        if (crypto_sign(signed_message, &signed_message_len,
+        reinterpret_cast<const unsigned char *>(udf), statbuf.st_size, sk) != 0)
+        {
+            fprintf(stderr, "Could not sign message\n");
+            delete[] signed_message;
+            return false;
+        }
+        return createDirectoryTree(pk, sk, signed_message, signed_message_len);
```

Comment:
You're leaking `signed_message` here. Better capture the output of `createDirectoryTree`, then `delete[] signed_message`, and then `return ret`.